src/notyf/shared/notyf.service.ts

```diff
@@ -1,77 +1,55 @@
-    private injector: Injector
-  ) {
-    this.toastContainerRef = this.createComponentRef(ToastContainerComponent);
-    this.toastContainerElement = this.getDOMElementFromComponentRef(this.toastContainerRef);
-    this.addChild(this.toastContainerElement);
-  }
-
-  setToastStyle(style: any) {
-    this.toastDynamicStyle = style;
-  }
-
-  setToastContainerStyle(styles: any) {
-    this.setDynamicStyles(styles, this.toastContainerRef);
+  constructor(private domService: DomService) {
+    this.toastContainerRef = this.domService.createComponentRef(ToastContainerComponent);
+    this.toastContainerElement = this.domService.getDomElementFromComponentRef(this.toastContainerRef);
+    this.domService.addChild(this.toastContainerElement);
   }
 
   alert(message: string) {
-    const toastRef = this.createComponentRef(ToastComponent);
+    const toastRef = this.domService.createComponentRef(ToastComponent);
     toastRef.instance.message = message;
-    this.setDynamicStyles(this.toastDynamicStyle, toastRef);
+    this.domService.setDynamicStyles(this._toastStyle, toastRef);
     toastRef.instance.type = ToastType.Alert;
-    const toastElement = this.getDOMElementFromComponentRef(toastRef);
-    this.addChild(toastElement, this.toastContainerElement);
-    this.destroyRef(toastRef, this.toastDelay);
+    const toastElement = this.domService.getDomElementFromComponentRef(toastRef);
+    this.domService.addChild(toastElement, this.toastContainerElement);
+    this.domService.destroyRef(toastRef, this._toastDelay);
+  }
+
+  get toastDelay(): number {
+    return this._toastDelay;
   }
 
-  private createComponentRef(component: any): ComponentRef<any> {
-    const componentRef = this.componentFactoryResolver
-      .resolveComponentFactory(component)
-      .create(this.injector);
-    this.appRef.attachView(componentRef.hostView);
-    return componentRef;
+  set toastDelay(toastDelay) {
+    this._toastDelay = toastDelay > 0 ? toastDelay : 0;
   }
 
-  private getDOMElementFromComponentRef(componentRef: ComponentRef<any>) {
-    return (componentRef.hostView as EmbeddedViewRef<any>)
-      .rootNodes[0] as HTMLElement;
+  get toastStyle(): {} {
+    return this._toastStyle;
   }
 
-  private addChild(child: HTMLElement, parent: HTMLElement = document.body) {
-    parent.appendChild(child);
+  set toastStyle(toastStyle: {}) {
+    this._toastStyle = toastStyle;
   }
 
-  private destroyRef(componentRef, delay) {
-    setTimeout(() => {
-      this.appRef.detachView(componentRef.hostView);
-      componentRef.destroy();
-    }, delay);
+  get toastContainerStyle(): {} {
+    return this._toastContainerStyle;
   }
 
-  private setDynamicStyles(styles: any, componentRef: ComponentRef<any>) {
-    for (const [styleName, styleValue] of Object.entries(styles)) {
-      componentRef.instance.renderer.setElementStyle(componentRef.instance.elementRef.nativeElement, styleName, styleValue);
-    }
+  set toastContainerStyle(toastContainerStyle: {}) {
```

Comment:
I don't use `any` because we know for sure the type is an object (`{}`). 

I think that for the sake of having a more clean code we can add the types of the object, but I don't think we need an `interface`, it just takes to add it as the parameter type:
```
set toastContainerStyle(toastContainerStyle: {[rule: string]: string}) {
}
```
BTW, I didn't know it was possible 😛